### DISCUSSION TITLE
Mobile App Wallet not launching from PWA

### DISCUSSION THREAD

**Discussion by ms-safiq** [reown-com/appkit#2834] [Q&A]
<img width="345" alt="Screenshot 2024-09-10 at 4 24 39 PM" src="https://github.com/user-attachments/assets/72187fbd-b920-4ae8-94c3-a46aaad262ab">

Hey ,

I have built wallet connect in angular ts web project. when i try to access from mobile browser, its simply launching the wallet connect and when i click on metamask, its not launching my metamask mobile app.  It stays on the same screen. How do i properly launch the mobile app when i launching from mobile browser? Basically its struck as seen in the image attached. its not launching any app chooser. Am i missing any configuration? Below is my code

`const projectId = environment.wcProjectId;

    const chainOptions: any = {
      chainId: Number(environment.chainId),
      name: environment.chainName,
      currency: environment.currency,
      explorerUrl: environment.explorerUrl,
      rpcUrl: environment.rpcUrl,
    };

    const metadata = {
      name: "Test",
      description: environment.selfDomain,
      url: environment.selfDomain,
      icons: [environment.selfDomain + "/assets/icons/logo.svg"],
    };

    const ethersConfig = defaultConfig({
      auth: {
        email: false,
        socials: [],
        showWallets: true,
        walletFeatures: true,
      },
      enableInjected: true,
      metadata,
    });

    this.wcModal = createWeb3Modal({
      ethersConfig,
      chains: [chainOptions],
      projectId,
      enableOnramp: true,
    });`

I am using Appkit ehters javascript library

@rtomas Please guide

**Comment by rtomas** [comment]
Hi! 
Please share a minimal reproducible example ?
So we can double check the issue.

Thanks

**Reply by ms-safiq** [reply]
Okay, Before that, if the wallet connect works in web application, will the same code work in mobile browser too without any changes in the code?

**Reply by ms-safiq** [reply]
Minimal Reproducible Sample

package.json

`"@web3modal/ethers": "5.1.6","ethers": "^6.13.2",`

Below is the method I call on a click of one simple button. Please input your own projectId. 

`initWalletConnect() {
    const projectId = environment.wcProjectId;

    const chainOptions: any = {
      chainId: 421614,
      name: "Arbitrum Sepolia",
      currency: "SepoliaETH",
      explorerUrl: "https://sepolia.arbiscan.io",
      rpcUrl: "https://sepolia-rollup.arbitrum.io/rpc",
    };

    const metadata = {
      name: "AppName",
      description: environment.selfDomain,
      url: environment.selfDomain,
      icons: [environment.selfDomain + "/assets/icons/logo.svg"],
    };

    const ethersConfig = defaultConfig({
      auth: {
        email: false,
        socials: [],
        showWallets: true,
        walletFeatures: true,
      },
      enableInjected: true,
      metadata,
    });

    this.wcModal = createWeb3Modal({
      ethersConfig,
      chains: [chainOptions],
      projectId,
      enableOnramp: true,
    });
    this.wcModal.open()
  }
`

Thats all. Now this code launches the wallet connect modal, but when i click on metamask ,its keep loading as in the attached image in my initial question. I was expecting that it will launch an app chooser. I am checking from the android device.

**Reply by ms-safiq** [reply]
@rtomas

**Reply by ms-safiq** [reply]
@rtomas Any updates would be really helpful

**Reply by ms-safiq** [reply]
@rtomas Basically the copy link is not generating any wc: format. What am i missing in the configuration?

**Reply by ms-safiq** [reply]
@rtomas Wallet connect QR is not generating when launching from mobile browser.

**Reply by rtomas** [reply]
would you mind sharing a repository for a functional minimal reproducible example ? so I can test what your problem is ?

if QR is not generating. please also share your console errors.
